app/helpers/practices_helper.rb

```diff
@@ -38,11 +38,197 @@ def duration_of_job_display(practice)
     durations.include?('permanent') ? 'Permanent' : "#{durations.map { |d| d.to_i }.sum} weeks" if durations.any?
   end
 
+  def fetch_page_view_leader_board(duration = "30", limit = 10)
+    page_view_leaders = []
+    sql = "select name, properties, count(properties) as count from ahoy_events where name = 'Practice show' and time >= now() - interval '{#{duration} days' group by name, properties order by count desc limit #{limit}"
+    records_array = ActiveRecord::Base.connection.execute(sql)
+
+    recs = records_array.values
+    recs.each do |rec|
+      practice_id = JSON.parse(rec[1])['practice_id']
+      leader = PracticeLeaderBoard.new
+      leader.practice_name = Practice.find(practice_id).name
+      leader.count = rec[2]
+      page_view_leaders << leader
+    end
+    page_view_leaders
+  end
+
+  def fetch_page_view_for_practice_count(practice_id, duration = "30")
+    page_view_leaders = []
+    sql = "select name, properties, count(properties) as count from ahoy_events where name = 'Practice show' and time >= now() - interval '{#{duration} days' group by name, properties"
+    records_array = ActiveRecord::Base.connection.execute(sql)
```

Comment:
rather than doing sql queries in this file can we move them as methods on the model?

Reply:
keeping this for v1